lib/improver/tests/optical_flow/test_AdvectField.py

```diff
@@ -0,0 +1,285 @@
+        result = self.dummy_plugin._advect_field(
+            self.data, self.grid_vel_x, self.grid_vel_y, 2., 0.)
+        self.assertIsInstance(result, np.ndarray)
+        self.assertArrayAlmostEqual(result, expected_output)
+
+    def test_partial(self):
+        """Test advection by a quarter of a grid point in the x direction and
+        one grid point in the y direction"""
+        expected_output = np.array([[0., 0., 0.],
+                                    [0., 2.75, 3.75],
+                                    [0., 1.75, 2.75],
+                                    [0., 0.75, 1.75]])
+        result = self.dummy_plugin._advect_field(
+            self.data, self.grid_vel_x, 2.*self.grid_vel_y, 0.5, 0.)
+        self.assertIsInstance(result, np.ndarray)
+        self.assertArrayAlmostEqual(result, expected_output)
+
+
+class Test_process(IrisTest):
+    """Test dimensioned cube data is correctly advected"""
+
+    def setUp(self):
+        """Set up plugin instance and a cube to advect"""
+        vel_x = set_up_xy_velocity_cube("advection_velocity_x")
+        vel_y = vel_x.copy(data=2.*np.ones(shape=(4, 3)))
+        vel_y.rename("advection_velocity_y")
+        self.plugin = AdvectField(vel_x, vel_y)
+        data = np.array([[2., 3., 4.],
+                         [1., 2., 3.],
+                         [0., 1., 2.],
+                         [0., 0., 1.]])
+        self.cube = iris.cube.Cube(
+            data, standard_name='rainfall_rate', units='mm h-1',
+            dim_coords_and_dims=[(self.plugin.y_coord, 0),
+                                 (self.plugin.x_coord, 1)])
+
+        # input time: [datetime.datetime(2018, 2, 20, 4, 0)]
+        self.time_coord = DimCoord(1519099200, standard_name="time",
+                                   units='seconds since 1970-01-01 00:00:00')
+        self.cube.add_aux_coord(self.time_coord)
+
+        self.timestep = datetime.timedelta(seconds=600)
+
+    def test_basic(self):
+        """Test output cube data is as expected"""
+        expected_output = np.array([[0., 0., 0.],
+                                    [0., 0., 0.],
+                                    [0., 2., 3.],
+                                    [0., 1., 2.]])
+        result = self.plugin.process(self.cube, self.timestep)
+        self.assertArrayAlmostEqual(result.data, expected_output)
+
+    def test_background_values(self):
+        """Test output cube data is padded as expected where source grid
+        points are out of bounds"""
+        expected_output = np.array([[-1., -1., -1.],
+                                    [-1., -1., -1.],
+                                    [-1., 2., 3.],
+                                    [-1., 1., 2.]])
+        result = self.plugin.process(self.cube, self.timestep, bgd=-1.)
+        self.assertArrayAlmostEqual(result.data, expected_output)
+
+    def test_time_step(self):
+        """Test outputs are OK for a time step with non-second components"""
```

Comment:
I'd prefer the test to have some non-out-of-bounds data in the result. Can you change the test to use a smaller time step?